core/vm/gas_table.go

```diff
@@ -101,8 +101,9 @@ var (
 func gasExtCodeSize(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 	usedGas := uint64(0)
 	slot := stack.Back(0)
+	address := slot.Bytes20()
 	if evm.chainRules.IsPrague {
-		usedGas += evm.TxContext.Accesses.TouchAddressOnReadAndComputeGas(slot.Bytes(), uint256.Int{}, trieUtils.CodeSizeLeafKey)
+		usedGas += evm.TxContext.Accesses.TouchAddressOnReadAndComputeGas(address[:], uint256.Int{}, trieUtils.CodeSizeLeafKey)
```

Comment:
This was the offending line that caused the new panic.

But I was wondering if L105-107 are really needed. This method is the one that calculates the dynamic gas. This touch on `CodeSizeLeafKey` also happens in `opExtCodeSize` (which I also fixed, since if not it would panic there too).

The dynamic gas method is called before the actual opcode execution, so that's why it panicked here. But I think we can avoid the touching here, and just rely on the opcode execution method (i.e: `opExtCodeSize`)?

Reply:
yes, and in fact this is fixed in my access witness rebase but since it's going to take a bit of time, better merge this asap.